### PR TITLE
Add MultiInterfaceMode to installation spec

### PR DESCRIPTION
**Pull request by Brian-McM**
## Description
This configures if multiple interfaces are allowed per pod and what configures them. We only allow multus for now.

## For PR author

- [ ] Tests for change.
- [ ] If changing pkg/apis/, run `make gen-files`
- [ ] If changing versions, run `make gen-versions`

## For PR reviewers

A note for code reviewers - all pull requests must have the following:

- [ ] Milestone set according to targeted release.
- [ ] Appropriate labels:
  - `kind/bug` if this is a bugfix.
  - `kind/enhancement` if this is a a new feature.
  - `enterprise` if this PR applies to Calico Enterprise only.

### CODE REVIEW

File: deploy/crds/operator_v1_installation_crd.yaml [hunk 1]
```diff
@@ -88,6 +88,14 @@ spec:
                     for pods on the Calico network. Default: 1410'
                   format: int32
                   type: integer
+                multiInterfaceMode:
+                  description: 'MultiInterfaceMode configures what will configure
+                    multiple interface per pod. Only valid for EE installations. Default:
```
Comment (tmjd) [inline deploy/crds/operator_v1_installation_crd.yaml:93]: I don't think EE is appropriate to use here, should probably be `Calico Enterprise`

File: pkg/render/node.go [hunk 2]
```diff
@@ -635,6 +635,12 @@ func (c *nodeComponent) cniEnvvars() []v1.EnvVar {
 			},
 		},
 	}
+
+	if c.cr.Spec.CalicoNetwork != nil {
```
Comment (tmjd) [inline pkg/render/node.go:639]: Should this be gated on Spec.Variant === Enterprise (like the others)?

Comment (Brian-McM) [reply pkg/render/node.go]: yes it should

**Comment by caseydavenport** [comment]
@tmjd double check I put the right milestone on this one!

**Pull request merged**